assets/js/components/Dashboard/NewPaMessage/AssociateAlertPage.tsx

```diff
@@ -0,0 +1,347 @@
+            }}
+            className="apply-button"
+          >
+            Apply
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    </div>
+  );
+};
+
+interface AssociateAlertsTableProps {
+  alerts: Alert[];
+  messageStateFilter: string;
+  serviceTypeFilter: string;
+  setAssociatedAlert: Dispatch<SetStateAction<Alert>>;
+  setShowAlertModal: Dispatch<SetStateAction<boolean>>;
+}
+
+const AssociateAlertsTable: ComponentType<AssociateAlertsTableProps> = ({
+  alerts,
+  messageStateFilter,
+  serviceTypeFilter,
+  setAssociatedAlert,
+  setShowAlertModal,
+}: AssociateAlertsTableProps) => {
+  const filterByActiveState = (alert: Alert) => {
+    const alertStart = new Date(alert.active_period[0].start);
+    const currentTime = new Date();
+    return messageStateFilter === "active"
+      ? alertStart <= currentTime
+      : alertStart > currentTime;
+  };
+
+  const filterByServiceType = (alert: Alert) => {
+    return serviceTypeFilter === "All"
+      ? true
+      : alert.affected_list.some((affected) =>
+          affected.includes(serviceTypeFilter.toLowerCase()),
+        );
+  };
+
+  const filteredAlerts = alerts.filter((alert) => {
+    return filterByActiveState(alert) && filterByServiceType(alert);
+  });
+
+  return (
+    <>
+      <table className="associate-alert-table">
+        <thead>
+          <tr>
+            <th className="associate-alert-table__message">Alert message</th>
+            <th className="associate-alert-table__id">ID</th>
+            <th className="associate-alert-table__start-end">Start-End</th>
+            <th className="associate-alert-table__last-modified">
+              Last modified
+            </th>
+            <th className="associate-alert-table__select"></th>
+          </tr>
+        </thead>
+        <tbody>
+          {filteredAlerts.length == 0 ? (
+            <tr>
+              <div className="associate-alert-table__empty">
+                There are no alerts of this type
+              </div>
+            </tr>
+          ) : (
+            filteredAlerts.map((alert: Alert) => {
+              return (
+                <AssociateAlertsTableRow
+                  key={alert.id}
+                  alert={alert}
+                  setAssociatedAlert={setAssociatedAlert}
+                  setShowAlertModal={setShowAlertModal}
+                />
+              );
+            })
+          )}
+        </tbody>
+      </table>
+    </>
+  );
+};
+
+interface AssociateAlertsTableRowProps {
+  alert: Alert;
+  setAssociatedAlert: Dispatch<SetStateAction<Alert>>;
+  setShowAlertModal: Dispatch<SetStateAction<boolean>>;
+}
+
+const AssociateAlertsTableRow: ComponentType<AssociateAlertsTableRowProps> = ({
+  alert,
+  setAssociatedAlert,
+  setShowAlertModal,
+}: AssociateAlertsTableRowProps) => {
+  const [start, end] = getAlertEarliestStartLatestEnd(alert.active_period);
+
+  const last_modified = moment(alert.updated_at).format("l LT");
+
+  return (
+    <tr
```

Comment:
Since you can click the row itself to select the alert, should it have a pointer cursor? I think it would help the user recognize that the row is clickable.

Reply:
Yeah I think it doesn't hurt. I'll add that in